pkg/bloomgateway/bloomgateway.go

```diff
@@ -161,6 +161,40 @@ func (g *Gateway) stopping(_ error) error {
 	return services.StopManagerAndAwaitStopped(context.Background(), g.serviceMngr)
 }
 
+func (g *Gateway) PrefetchBloomBlocks(_ context.Context, req *logproto.PrefetchBloomBlocksRequest) (*logproto.PrefetchBloomBlocksResponse, error) {
+	refs, err := decodeBlockKeys(req.Blocks)
+	if err != nil {
+		return nil, err
+	}
+
+	bqs, err := g.bloomStore.FetchBlocks(
+		// We don't use the ctx passed to the handler since its canceled when the handler returns
+		context.Background(),
```

Comment:
Not passing a timeout here since you always need to call the cancel function when returning from the function, which would trigger the download to stop. This is not an issue since when the gateway stops, the queue is closed and any pending block download is stopped.